x-pack/plugins/enterprise_search/server/lib/indices/fetch_indices.ts

```diff
@@ -34,12 +34,12 @@ export const fetchIndices = async (
   indexPattern: string,
   returnHiddenIndices: boolean,
   includeAliases: boolean,
-  alwaysShowSearchPattern?: 'search-'
+  alwaysShowPattern?: AlwaysShowPattern
 ): Promise<ElasticsearchIndexWithPrivileges[]> => {
   // This call retrieves alias and settings information about indices
-  // If we provide an override pattern with alwaysShowSearchPattern we get everything and filter out hiddens.
+  // If we provide an override pattern with alwaysShowPattern we get everything and filter out hiddens.
   const expandWildcards: ExpandWildcard[] =
-    returnHiddenIndices || alwaysShowSearchPattern ? ['hidden', 'all'] : ['open'];
+    returnHiddenIndices || alwaysShowPattern?.alias_pattern ? ['hidden', 'all'] : ['open'];
```

Comment:
In the case where there is an `index_pattern` passed but no `alias_pattern` passed this will default to `expandWildcards: ['open']` and not include hidden indices.

I am aware we are calling `fetchIndices` on certain occasions, and currently there is no call like this yet, but it would break easily if it happens.

We could simply change this statement a little bit as following.
`returnHiddenIndices || (alwaysShowPattern?.alias_pattern || alwaysShowPattern?.index_pattern) ? ['hidden', 'all'] : ['open'];`

Reply:
That's a good catch. I missed to notice. Will change.